activesupport/lib/active_support/testing/assertionless_tests.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module Testing
+    # Allows to configure the behavior when the test case has no assertions in it.
+    # This is helpful in detecting broken tests that do not perform intended assertions.
+    module AssertionlessTests # :nodoc:
+      def self.prepended(klass)
+        klass.singleton_class.attr_reader :assertionless_tests_behavior
+        klass.extend ClassMethods
+        klass.assertionless_tests_behavior = :ignore
+      end
+
+      module ClassMethods
+        def assertionless_tests_behavior=(behavior)
+          @assertionless_tests_behavior =
+            case behavior
+            when :ignore
+              nil
+            when :log
+              logger =
+                if defined?(Rails.logger) && Rails.logger
+                  Rails.logger
```

Comment:
Not a big fan of checking for Rails existence here. Might make more sense to introduce `ActiveSupport.logger`.

Reply:
This is how it is done in another place too https://github.com/rails/rails/blob/f04c59b9ff96a5b7b5b3c498e758055081d3bc08/activesupport/lib/active_support/deprecation/behaviors.rb#L25-L33

Maybe introduce that change separately?